File: 20_sezona-2020-21/01_KV1-2020-21/04_cvrsti_brojevi/cvrsti_brojevi-TadijaSebez.py
s = input()
dp = [[0 for _ in range(10)] for _ in range(1024)]
best = [0 for _ in range(1024)]
for c in s:
	o = ord(c)-ord('0')
	for mask in range(1024):
		if ((mask>>o)&1) == 1:
			dp[mask][o] = max(dp[mask][o], best[mask^(1<<o)])+1
			best[mask] = max(best[mask], dp[mask][o])
ans = 0
for i in best:
	ans = max(ans, i)
print(len(s)-ans)

File: 20_sezona-2020-21/01_KV1-2020-21/04_cvrsti_brojevi/cvrsti_brojevi-TadijaSebez.cpp
#include <bits/stdc++.h>
using namespace std;
void ckmn(int&a,int b){a=min(a,b);}
void ckmx(int&a,int b){a=max(a,b);}
const int N=100050;
const int M=1<<10;
const int inf=1e9+7;
char s[N];
int dp[10][M],best[M];
int main(){
	scanf("%s",s+1);
	int n=strlen(s+1);
	fill(dp[0],dp[0]+M*10,-inf);
	fill(best,best+M,-inf);
	for(int i=0;i<10;++i)dp[i][1<<i]=0,best[1<<i]=0;
	for(int i=1;i<=n;++i){
		int c=s[i]-'0';
		for(int mask=M-1;mask>=0;--mask){
			if(mask>>c&1){
				dp[c][mask]=max(dp[c][mask],best[mask^(1<<c)])+1;
				ckmx(best[mask],dp[c][mask]);
			}
		}
	}
	int ans=-inf;
	for(int i=0;i<M;i++)ckmx(ans,best[i]);
	printf("%i\n",n-ans);
	return 0;
}